### ISSUE TITLE
clang-tidy 17.0.2 check google-readability-casting false-positive result for std::vector constructor call

### ISSUE THREAD

**Issue by pkl97** [llvm/llvm-project#69954]
This code triggers an incorrect cast warning for a std::vector constructor call in clang-tidy 17.0.2:
```c++
#include <vector>

template<typename T>
class MatrixOperation
{
public:
    void performOperation();
private:
    std::vector<double> m_x;
};

template<typename T>
void MatrixOperation<T>::performOperation()
{
    const std::size_t n=m_x.size();
    std::vector<std::vector<double>> matrix(n);
    for(size_t i=0; i<n; ++i)
        matrix[i]=std::vector<double>(n);
}

int main()
{
    MatrixOperation<std::string>().performOperation();
    return 0;
}
```
The emitted clang-tidy warning:
```console
main.cpp:18:19: warning: C-style casts are discouraged; use static_cast/const_cast/reinterpret_cast [google-readability-casting]
   18 |         matrix[i]=std::vector<double>(n);
      |                   ^
```

**Comment by chrchr-github** [comment]
Slightly reduced: https://godbolt.org/z/1Kr5d96E6

**Comment by PiotrZSL** [comment]
Duplicate of https://github.com/llvm/llvm-project/issues/57959
